src/GNDStk/v1.9/containers/Axis/test/Axis.test.cpp

```diff
@@ -9,55 +9,66 @@
 using namespace njoy::GNDStk;
 using Axis = v1_9::containers::Axis;
 
-Node defaultChunk();
-void verifyDefaultChunk( const Axis& );
-Node chunk();
+std::string chunk();
 void verifyChunk( const Axis& );
-Node invalidName();
+std::string invalidName();
 
 SCENARIO( "Axis" ) {
 
-  GIVEN( "valid data for a Axis" ) {
+  GIVEN( "valid data for the component" ) {
 
-    WHEN( "the default constructor is used" ) {
+    std::string string = chunk();
 
-      Axis chunk;
-
-      THEN( "an Axis can be constructed and members can be tested" ) {
-
-        verifyDefaultChunk( chunk );
-      } // THEN
-    } // WHEN
-
-    WHEN( "the data is given explicitly" ) {
+    WHEN( "autogenerated constructor: the data is given explicitly" ) {
 
       unsigned int index = 1;
       std::string label = "energy_in";
       std::string unit = "eV";
 
       Axis chunk( index, label, unit );
 
-      THEN( "an Axis can be constructed and members can be tested" ) {
+      THEN( "the component can be constructed and members can be tested" ) {
 
         verifyChunk( chunk );
       } // THEN
+
+      THEN( "it can be written in XML" ) {
+
+        std::ostringstream out;
+        XML( Node( chunk ) ).write( out, false );
+
+        CHECK( out.str() == string );
+      } // THEN
     } // WHEN
 
     WHEN( "the data is constructed from a node" ) {
 
-      Node core = chunk();
-      Axis chunk( core );
+      Node node;
+      node << chunk();
+      Axis chunk( node );
 
-      THEN( "an Axis can be constructed and members can be tested" ) {
+      THEN( "the component can be constructed and members can be tested" ) {
```

Comment:
I see how genericizing the terminology like this, will help in the long terms when we have lots of tests for lots of classes.

Reply:
Indeed. I'm already thinking of at least autogenerating a skeleton unit test file once I have converged to a specific style.